### PR TITLE
Change BaseType to TypeTag, and allow multiples

### PR DESCRIPTION
**Pull request by koosemose**
Per further discussion in #31, BaseType was changed to TypeTag (to more closely match new functionality), and now allows multiples (previously it would be set to the last one found). Added bool HasTypeTag(String) method to separate usage from implementation (future-proofing for any potential changes in the future).

### CODE REVIEW

**Comment by WardBenjamin** [comment]
Looks great. I think this a good solution that leaves modding and new items open.

**Comment by koosemose** [comment]
Thanks, had an extended discussion with @lameox that helped further refine the original idea. It also has the added side-effect of things not erroring out if trying to replace a furniture that doesn't have a typeTag (or anything else that wants to look for a match)

File: Assets/Scripts/Models/Furniture.cs [hunk 1]
```diff
@@ -183,7 +184,7 @@ protected Furniture(Furniture other)
     {
         this.objectType = other.objectType;
         this.Name = other.Name;
-        this.baseType = other.baseType;
+        this.typeTags = other.typeTags;
```
Comment (henninglive) [inline Assets/Scripts/Models/Furniture.cs]: Is this correct? shouldn't you copy the hashset?

**Comment by koosemose** [comment]
You're probably right, I overlooked that, and even though that works in the current implementation since new typeTags are only added to the prototype when it's created, I'll get that sorted out in case someone later adds a way to directly modify a Furniture's typeTags (such as maybe a way to mark a Furniture as broken)

**Comment by koosemose** [comment]
Alright, that should fix any potential problems.

**Comment by henninglive** [comment]
When is the copy constructor called anyway? what its purpose?

**Comment by koosemose** [comment]
When you're creating an actual furniture from a prototype. In case you're not familiar with the setup, it reads the values from XML into a prototype furniture, then when an actual piece of Furniture is being built it uses the copies the prototype.

File: Assets/Scripts/Models/Furniture.cs [hunk 2]
```diff
@@ -183,7 +184,7 @@ protected Furniture(Furniture other)
     {
         this.objectType = other.objectType;
         this.Name = other.Name;
-        this.baseType = other.baseType;
+        this.typeTags = new HashSet<string>(other.typeTags);
```
Comment (relinu) [inline Assets/Scripts/Models/Furniture.cs]: @henninglive the copy constructor is called here,  it copies everything inside other.typeTags into this.typeTags

Comment (henninglive) [inline Assets/Scripts/Models/Furniture.cs]: I was wondering about the Furniture copy constructor, but @koosemose explained it to me.

**Comment by RodRitter** [comment]
Nice change. Tested and it's working fine.

**Pull request merged**